scripts/gen_help_html.lua

```diff
@@ -16,42 +18,23 @@
 --
 -- NOTES:
 --   * gen() and validate() are the primary entrypoints. validate() only exists because gen() is too
---   slow (~1 min) to run in per-commit CI.
+--     slow (~1 min) to run in per-commit CI.
 --   * visit_node() is the core function used by gen() to traverse the document tree and produce HTML.
 --   * visit_validate() is the core function used by validate().
 --   * Files in `new_layout` will be generated with a "flow" layout instead of preformatted/fixed-width layout.
---
--- parser bugs:
---  * Should NOT be code_block:
--- 	  tab:xy	The 'x' is always used, then 'y' as many times as will
--- 			fit.  Thus "tab:>-" displays:
--- 				>
--- 				>-
--- 				>--
--- 				etc.
---
--- 	  tab:xyz	The 'z' is always used, then 'x' is prepended, and
--- 			then 'y' is used as many times as will fit.  Thus
--- 			"tab:<->" displays:
--- 				>
--- 				<>
--- 				<->
--- 				<-->
--- 				etc.
---  * Should NOT be a "headline". Perhaps a "table" (or just "line").
---        expr5 and expr6						*expr5* *expr6*
---        ---------------
---        expr6 + expr6   Number addition, |List| or |Blob| concatenation	*expr-+*
---        expr6 - expr6   Number subtraction				*expr--*
---        expr6 . expr6   String concatenation				*expr-.*
---        expr6 .. expr6  String concatenation				*expr-..*
 
 local tagmap = nil
 local helpfiles = nil
-local invalid_tags = {}
+local invalid_links = {}
+local invalid_urls = {}
+local invalid_spelling = {}
+local spell_dict = {
+  Neovim = 'Nvim',
+  NeoVim = 'Nvim',
+  neovim = 'Nvim',
+  lua = 'Lua',
+}
```

Comment:
experimenting with other lint stuff here (TODO: obviously `scripts/gen_help_html.lua` is not really the place for this kind of thing, will eventually revisit this)